webui/react/src/pages/WorkspaceDetails/WorkspaceProjects.tsx

```diff
@@ -98,13 +98,25 @@ const WorkspaceProjects: React.FC<Props> = ({ workspace, id, pageRef }) => {
     } finally {
       setIsLoading(false);
     }
-  }, [canceler.signal, id, workspace, settings]);
+  }, [canceler.signal, id, workspace?.archived, settings]);
+
+  const previousId = usePrevious(id, undefined);
 
   useEffect(() => {
     setIsLoading(true);
-    fetchProjects().then(() => setIsLoading(false));
+    fetchProjects();
+  }, [fetchProjects]);
+
+  useEffect(() => {
+    /**
+     * as we don't unmount this component whenever we change from workspace to workspace
+     * this effect simply "re-runs" the settings to verify all applied settings and checks
+     * like setting active settings into the url, etc
+     */
+
+    if (previousId !== undefined && id !== previousId) updateSettings({});
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [id]);
```

Comment:
i understand `previousId` can be out of array, but dont we need `updateSettings` in dep array?